### PR TITLE
Add void return type hint

### PR DESCRIPTION
**Pull request by greg0ire**
## Changelog

```markdown
### Added
- `void` return type hint wherever applicable

```

### CODE REVIEW

File: src/Exporter.php [hunk 1]
```diff
@@ -57,7 +57,7 @@ public function getResponse(string $format, string $filename, SourceIteratorInte
         }
         $writer = $this->writers[$format];
 
-        $callback = function () use ($source, $writer) {
+        $callback = function () use ($source, $writer): void {
```
Comment (core23) [inline src/Exporter.php]: This is needed? oO

Comment (greg0ire) [reply src/Exporter.php]: If you mean "this works?", yes it works :) (why not?). If you mean "Is this necessary?" none of this PR is, it just makes things safer.

Comment (core23) [reply src/Exporter.php]: I can understand this stuff for the api methods (protected and public things). But for internal functions? I don't think that this is useful at all.

Comment (greg0ire) [reply src/Exporter.php]: It doesn't hurt though, and I use php-cs-fixer for this, and I'd hate to make an exception for this kind of stuff each time I run it. Also I'd argue it's good to have type hinting for internal functions: it helps your IDE understanding what happens, and if you do wrong things it crashes early.

File: tests/ExporterTest.php [hunk 2]
```diff
@@ -30,7 +30,7 @@ public function testFilter()
         $exporter->getResponse('foo', 'foo', $source);
     }
 
-    public function testConstructorRejectsNonTypedWriters()
+    public function testConstructorRejectsNonTypedWriters(): void
     {
         $this->setExpectedException(
             version_compare(PHP_VERSION, '7.0.0', '<') ? 'PHPUnit_Framework_Error' : 'TypeError',
```
Comment (jordisala1991) [inline tests/ExporterTest.php]: PHP5 line detected!

Comment (greg0ire) [reply tests/ExporterTest.php]: @jordisala1991 fixed

**Comment by greg0ire** [comment]
@sonata-project/contributors please merge

**Pull request merged**